piyushgarg-dev/piyushgargdev-nextjs
# FIX :  Unknown Error Occurred on Contact Form Submission on Homepage

## What does this PR do?

Fixes the Contact form feature by using googleapi library.

Fixes #119 

<img width="1325" alt="Screenshot 2023-06-16 at 2 00 34 AM" src="https://github.com/piyushgarg-dev/piyushgargdev-nextjs/assets/75656830/9cf2fb01-c36e-4014-a239-14958b2073df">
<img width="712" alt="Screenshot 2023-06-16 at 2 18 54 AM" src="https://github.com/piyushgarg-dev/piyushgargdev-nextjs/assets/75656830/8d2d04fe-2a9b-4bbd-b301-08b0fb0fe9e4">


## Type of change in Local environment before Testing
- Download ````google-credentials.json aka(secret.json)```` from your Google Cloud Platform.
- Save that ````secret.json```` file in ````config```` folder.
- Public and Copy the Spreadsheet id and fill in ````.env````.
- Good to go.

## Steps to Test/Reproduce?
To test the fix for the "Unknown Error Occurred" issue, follow these steps :
- Go to the homepage of the website.
- Fill out the contact form with valid information.
- Submit the form.
- Verify that the form submission is successful without any error messages.
